lib/logging/logger_factory.js

```diff
@@ -19,15 +19,33 @@
 // THE SOFTWARE.
 'use strict';
 
-function gossipLogger(ringpop, msg, meta) {
-    var logLevel = ringpop.config.get('gossipLogLevel');
-    if (logLevel !== 'info' && logLevel !== 'debug') {
-        logLevel = 'debug';
-    }
+var ModuleLogger = require('./module_logger.js');
 
-    ringpop.logger[logLevel](msg, meta);
+function LoggerFactory(opts) {
+    this.ringpop = opts.ringpop;
+    this.loggers = {}; // indexed by name
 }
 
-module.exports = {
-    gossipLogger: gossipLogger
+LoggerFactory.prototype.getLogger = function getLogger(name) {
+    if (this.loggers[name]) {
+        return this.loggers[name];
+    }
+
+    switch (name) {
+        case 'gossip':
+            this.loggers[name] = new ModuleLogger(this.ringpop,
+                'gossipLogLevel');
+            break;
+        case 'joiner':
+            this.loggers[name] = new ModuleLogger(this.ringpop,
+                'joinerLogLevel');
+            break;
+        default:
+            this.loggers[name] = new ModuleLogger(this.ringpop,
```

Comment:
on second read (and reading further down), could this be better handled as a fallthrough, so that I can set foobarLogLevel in my config, and have it still work?